File: SDKs/amazonaws.com/detective/2018-10-26/go/pkg/models/operations/createmembers.go
package operations

import (
"openapi/pkg/models/shared")

type CreateMembersHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    
}

type CreateMembersRequestBody struct {
    Accounts []shared.Account `json:"Accounts"`
    DisableEmailNotification *bool `json:"DisableEmailNotification,omitempty"`
    GraphArn string `json:"GraphArn"`
    Message *string `json:"Message,omitempty"`
    
}

type CreateMembersRequest struct {
    Headers CreateMembersHeaders 
    Request CreateMembersRequestBody `request:"mediaType=application/json"`
    
}

type CreateMembersResponse struct {
    ContentType string 
    CreateMembersResponse *shared.CreateMembersResponse 
    InternalServerException *interface{} 
    ResourceNotFoundException *interface{} 
    ServiceQuotaExceededException *interface{} 
    StatusCode int64 
    ValidationException *interface{} 
    
}

File: SDKs/amazonaws.com/detective/2018-10-26/typescript/src/sdk/models/operations/createmembers.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class CreateMembersHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;
}


export class CreateMembersRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=Accounts", elemType: shared.Account })
  accounts: shared.Account[];

  @SpeakeasyMetadata({ data: "json, name=DisableEmailNotification" })
  disableEmailNotification?: boolean;

  @SpeakeasyMetadata({ data: "json, name=GraphArn" })
  graphArn: string;

  @SpeakeasyMetadata({ data: "json, name=Message" })
  message?: string;
}


export class CreateMembersRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: CreateMembersHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: CreateMembersRequestBody;
}


export class CreateMembersResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  createMembersResponse?: shared.CreateMembersResponse;

  @SpeakeasyMetadata()
  internalServerException?: any;

  @SpeakeasyMetadata()
  resourceNotFoundException?: any;

  @SpeakeasyMetadata()
  serviceQuotaExceededException?: any;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  validationException?: any;
}
